### PR TITLE
topdirs: define printer types manually

### PR DESCRIPTION
**Pull request by gasche**
Note: this commit does not touch the debugger and in particular it does not remove the new code in Genprintval, this is more of a proof of concept.

### CODE REVIEW

File: toplevel/topdirs.ml [hunk 1]
```diff
@@ -185,6 +185,29 @@ let _ = add_directive "mod_use" (Directive_string (with_error_fmt dir_mod_use))
...
+let type_arrow ta tb =
+  Ctype.newty (Tarrow (Asttypes.Nolabel, ta, tb, commu_var ()))
+
+let type_formatter () =
+  let format =
+    Env.lookup_module_path
+      ~use:false ~loc:Location.none ~load:false
+      (Longident.Lident "Format")
+      !toplevel_env in
+  Ctype.newconstr (Path.Pdot(format, "formatter")) []
```
Comment (Octachron) [inline toplevel/topdirs.ml:193]: I would propose to forge the `format` identifier with `Ident.create_persistent`.

Comment (gasche) [reply toplevel/topdirs.ml:193]: Ah, yes.

Comment (gasche) [reply toplevel/topdirs.ml:193]: Rebased.

**Comment by shindere** [comment]
Plenty of thanks, @gasche.

So did you update your PR with @Octachron's suggestion?

Did you run the testsuite to make sure your proposal not aonly compiles
but actually works?

**Comment by gasche** [comment]
For the record, the change I made in response to @Octachron's suggestion was

```diff
 let type_formatter () =
-  let format =
-    Env.lookup_module_path
-      ~use:false ~loc:Location.none ~load:false
-      (Longident.Lident "Format")
-      !toplevel_env in
+  let format = Path.Pident(Ident.create_persistent "Format") in
   Ctype.newconstr (Path.Pdot(format, "formatter")) []
```

**Comment by shindere** [comment]
Thanks! It's a nice improvement!

**Pull request closed without merging**